packages/cli/src/execute/handler.ts

```diff
@@ -33,16 +66,64 @@ const executeHandler = async (options: ExecuteOptions, logger: Logger) => {
     }
   }
 
-  const state = await loadState(options, logger);
+  let customStart;
+  let customEnd;
+
+  // Handle start, end and only
+  if (options.only) {
```

Comment:
This interaction here - not just the working out which step to run, but how it is fed to the runtime AND reported to the user - is not well tested.

It's taken quite a while to work out a good UX.

I don't want to spend much more time on this but maybe I'll see if I can find a good way to test this. I kinda worry that it'll take half a day to get a good suite of tests on it - not worth it right now.